src/PowerShellEditorServices/Workspace/ScriptFile.cs

```diff
@@ -311,12 +311,34 @@ public void ValidatePosition(BufferPosition bufferPosition)
         /// <param name="column">The 1-based column to be validated.</param>
         public void ValidatePosition(int line, int column)
         {
-            int maxLine = this.FileLines.Count;
+            ValidatePosition(line, column, isInsertion: false);
+        }
+
+        /// <summary>
+        /// Throws ArgumentOutOfRangeException if the given position is outside
+        /// of the file's buffer extents.
+        /// </summary>
+        /// <param name="line">The 1-based line to be validated.</param>
+        /// <param name="column">The 1-based column to be validated.</param>
+        /// <param name="isInsertion">If true, the position to validate is for an applied change.</param>
+        public void ValidatePosition(int line, int column, bool isInsertion)
+        {
+            int maxLine = isInsertion ? this.FileLines.Count + 1 : this.FileLines.Count;
```

Comment:
I would add a code comment above this to make sure someone else doesn't think they see an off-by-one error with the `this.FileLines.Count + 1` and attempt to fix it.  :-)